modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -322,14 +322,14 @@ inline std::vector<int64_t> toORT(const cv::MatSize &sz) {
 inline void preprocess(const cv::Mat& src,
                        const cv::gimpl::onnx::TensorInfo& ti,
                              cv::Mat& dst) {
-    GAPI_Assert(src.depth() == CV_32F || src.depth() == CV_8U);
+    GAPI_Assert(src.depth() == CV_32F || src.depth() == CV_8U || src.depth() == CV_32S);
     // CNN input type
     const auto type = toCV(ti.type);
-    if (src.depth() == CV_32F) {
+    if (src.depth() == CV_32F || src.depth() == CV_32S) {
```

Comment:
`preprocess` should be applied only if `CV_8U` is present - image case

This is how I would expect it to work:
```
// no assert here
if (src.depth() != CV_8U) {
    // Just pass the tensor as-is logic
} else {
    // 8U input: full preprocessing path
}
```